config/initializers/diaspora_federation.rb

```diff
@@ -61,5 +35,30 @@
 
       person_entity.save!
     end
+
+    on :fetch_private_key_by_id do |diaspora_id|
+      key = Person.where(diaspora_handle: diaspora_id).joins(:owner).pluck(:serialized_private_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :fetch_private_key_by_post_guid do |guid|
+      key = Post.where(guid: guid).joins(author: :owner).pluck(:serialized_private_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :fetch_public_key_by_id do |diaspora_id|
+      key = Person.where(diaspora_handle: diaspora_id).pluck(:serialized_public_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :fetch_public_key_by_post_guid do |guid|
+      key = Post.where(guid: guid).joins(:author).pluck(:serialized_public_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :post_author_is_local? do |guid|
+      id_array = Post.where(guid: guid).joins(:author).pluck(:owner_id)
+      !id_array.first.nil? unless id_array.empty?
```

Comment:
Wouldn't something like `Post.joins(:author).where(guid: guid).not(owner_id: nil).exists?` work?